Asynchronous deletion of records #190

ItIsJordan opened the pull request:
Implements asynchronous deletion of records and adds accompanying celery task.

Closes #190.

ItIsJordan committed:
Asynchronous submission deletion

Updated submission deletion to be asynchronous. Added a new accompanying celery task.

ItIsJordan committed:
Asynchronous submission deletion

Updated submission deletion to be asynchronous. Added a new accompanying celery task.

GraemeWatt reviewed (changes requested):
Thanks, I've tested and this works well.  Couple of requests for improvements:

- In `delete-widget.html` change text from ["Submission deleted."](https://github.com/HEPData/hepdata/blob/fb2646e365a9fe7145e8a4a2d7aebb46fbf3adf6/hepdata/modules/dashboard/templates/hepdata_dashboard/delete-widget.html#L43) to the more accurate "Submission will be deleted." (i.e. future tense since deletion is now asynchronous).
- Asynchronous deletion causes a potential new problem for deletion of large records like https://www.hepdata.net/record/ins1866951 which take more than 5 seconds to delete.  After 5 seconds the Dashboard/Sandbox is refreshed, but the record remains on the Dashboard/Sandbox since it has not been deleted yet.  If I wait a while (until the record has been deleted), then click the delete icon again without refreshing the page, I get stuck on the "Delete Submission" dialogue and the browser console shows an error message "Failed to load resource: the server responded with a status of 500 (INTERNAL SERVER ERROR)".  The terminal shows an exception `AttributeError: 'NoneType' object has no attribute 'version'` from [this line](https://github.com/HEPData/hepdata/blob/fb2646e365a9fe7145e8a4a2d7aebb46fbf3adf6/hepdata/modules/dashboard/views.py#L190).  I think we need to check that the submission exists (i.e. use `if submission:` before that line), `else` return a JSON error message like "This submission has already been deleted." that will be displayed to the user, then refresh the Dashboard/Sandbox.

Could you please try to address these two points?

ItIsJordan committed:
Merge branch 'asynchronous-deletion' of https://github.com/HEPData/hepdata into asynchronous-deletion

ItIsJordan committed:
Update views.py

Now verifies submission existence before attempting deletion, returns error if record exists.

ItIsJordan committed:
Change delete widget text and update tests

Changes the text of the deletion widget and updates relevant tests to use the new text.

ItIsJordan committed:
Update views.py

Changed incorrect 'error' to 'message' to enable message output in delete_submission.

GraemeWatt committed:
dashboard: improvements to delete widget

* Display new "delete-failure" window if submission already deleted.
